gprofiler/merge.py

```diff
@@ -332,6 +333,46 @@ def _enrich_and_finalize_stack(
     return f"{enrich_data.application_prefix}{enrich_data.container_prefix}{stack} {count}"
 
 
+def _merge_metadata(primary, secondary):
```

Comment:
This is fairly generic. Let's move to utils and keep with generic name `merge_dicts`? Also names like `source` and `dest` are more clear IMO, I liked them [here](https://stackoverflow.com/a/20666342/797390)